src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -446,5 +448,55 @@ public static bool RequiresToList(CSharpType from, CSharpType to)
 
             return to.FrameworkType == typeof(IReadOnlyList<>) || to.FrameworkType == typeof(IList<>);
         }
+
+        public CSharpType? GetCsharpType(ITypeSymbol symbol)
+        {
+            var stack = new Stack<string>();
+            var type = UnWrapType(symbol, stack);
+
+            while (stack.TryPop(out var wrappedType))
+            {
+                switch (wrappedType)
+                {
+                    case "ArmOperation":
+                        type = new CSharpType(typeof(ArmOperation<>), type!);
+                        break;
+                    case "Task":
+                        type = new CSharpType(typeof(Task<>), type!);
+                        break;
+                    case "AsyncPageable":
+                        type = new CSharpType(typeof(AsyncPageable<>), type!);
+                        break;
+                    case "Pageable":
+                        type = new CSharpType(typeof(Pageable<>), type!);
+                        break;
+                    case "Response":
+                        type = new CSharpType(typeof(Response<>), type!);
+                        break;
+                }
+            }
+            return type;
+        }
+
+        private CSharpType? UnWrapType(ITypeSymbol symbol, Stack<string> wrapers)
+        {
+            var namedTypeSymbol = symbol as INamedTypeSymbol;
+            if (namedTypeSymbol == null)
+            {
+                throw new InvalidCastException($"Unexpected type {symbol}");
+            }
+
+            if (namedTypeSymbol.TypeArguments.Any())
+            {
+                wrapers.Push(namedTypeSymbol.Name);
+                return UnWrapType(namedTypeSymbol.TypeArguments[0], wrapers);
+            }
```

Comment:
Handled multiple arguments and merged it to the original `TryCreateType` method, but unit testing for INamedTypeSymbol to CSharpType is not an easy task, the original method does not have UT either. This could be a new task requiring more time.

Reply:
Besides, we have the existing test projects to guard this logic, found an issue of it from `TypesSchemaMapping`

Reply:
This logic is not needed for ModelFactory until dealing with Resource, postpone the change to later PR instead.